test/e2e/volumes.go

```diff
@@ -0,0 +1,332 @@
+// and check that the pod sees the data from the server pod.
+func testVolumeClient(client *client.Client, config VolumeTestConfig, volume api.VolumeSource, expectedContent string) {
+	By(fmt.Sprint("starting ", config.prefix, " client"))
+	podClient := client.Pods(config.namespace)
+
+	clientPod := &api.Pod{
+		TypeMeta: api.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1beta3",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name: config.prefix + "-client",
+			Labels: map[string]string{
+				"role": config.prefix + "-client",
+			},
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:  config.prefix + "-client",
+					Image: "gcr.io/google_containers/nginx:1.7.9",
+					Ports: []api.ContainerPort{
+						{
+							Name:          "web",
+							ContainerPort: 80,
+							Protocol:      api.ProtocolTCP,
+						},
+					},
+					VolumeMounts: []api.VolumeMount{
+						{
+							Name:      config.prefix + "-volume",
+							MountPath: "/usr/share/nginx/html",
+						},
+					},
+				},
+			},
+			Volumes: []api.Volume{
+				{
+					Name:         config.prefix + "-volume",
+					VolumeSource: volume,
+				},
+			},
+		},
+	}
+	if _, err := podClient.Create(clientPod); err != nil {
+		Failf("Failed to create %s pod: %v", clientPod.Name, err)
+	}
+	expectNoError(waitForPodRunningInNamespace(client, clientPod.Name, config.namespace))
+
+	By("reading a web page from the client")
+	body, err := client.Get().
+		Namespace(config.namespace).
+		Prefix("proxy").
+		Resource("pods").
+		Name(clientPod.Name).
+		DoRaw()
+	expectNoError(err, "Cannot read web page: %v", err)
+	Logf("body: %v", string(body))
+
+	By("checking the page content")
+	Expect(body).To(ContainSubstring(expectedContent))
+}
+
+var _ = Describe("Volume", func() {
+	clean := true // If 'false', the test won't clear its namespace (and pods and services) upon completion. Useful for debugging.
+
+	// filled in BeforeEach
+	var c *client.Client
+	var namespace *api.Namespace
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+		By("Building a namespace api object")
+		namespace, err = createTestingNS("volume", c)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if clean {
+			if err := c.Namespaces().Delete(namespace.Name); err != nil {
+				Failf("Couldn't delete ns %s", err)
+			}
+		}
+	})
+
+	////////////////////////////////////////////////////////////////////////
+	// NFS
+	////////////////////////////////////////////////////////////////////////
+
+	// Marked with [Skipped] to skip the test by default (see driver.go),
+	// the test needs privileged containers, which are disabled by default.
+	// Run the test with "go run hack/e2e.go ... --ginkgo.focus=Volume"
+	Describe("[Skipped] NFS", func() {
+		It("should be montable", func() {
+			config := VolumeTestConfig{
+				namespace:   namespace.Name,
+				prefix:      "nfs",
+				serverImage: "gcr.io/google_containers/volume-nfs",
```

Comment:
@saad-ali, @jsafrane, @childsb
So both the nfs and gluster tests still fail b/c tag is missing.  

\# docker pull gcr.io/google_containers/volume-nfs
Pulling repository gcr.io/google_containers/volume-nfs
FATA[0001] Tag latest not found in repository gcr.io/google_containers/volume-nfs

\# docker pull gcr.io/google_containers/volume-nfs:0.1
Pulling repository gcr.io/google_containers/volume-nfs
7faef5ba165a: Download complete 
e9e06b06e14c: Download complete 
a82efea989f9: Download complete 
37bea4ee0c81: Download complete 
07f8e8c5e660: Download complete 
a61fbd375433: Download complete 
36ed446fd761: Download complete 
5c06e7e161bd: Download complete 
087a9dde26b9: Download complete 
56da8d2b187a: Download complete 
8cb003e95f03: Download complete 
9570a95e2d95: Download complete 
Status: Downloaded newer image for gcr.io/google_containers/volume-nfs:0.1